### PR TITLE
[PFS-207] Add Picker Protos

### PR DESCRIPTION
**Pull request by FahadBSyed**
This PR adds picker protos and their auto-generated code changes. The implementations of the pickers will be in a follow-up PR.

### CODE REVIEW

File: src/pfs/pfs.proto [hunk 1]
```diff
@@ -123,7 +151,40 @@ message Commit {
...
+// Commits can be picked relatively from some other commit like a parent or start of branch.
+// Alternatively, they can be picked via their global Id, which is composed of a repo picker and an id.
+// Picker messages should only be used as request parameters.
+message CommitPicker {
+  message CommitByGlobalId {
+    RepoPicker repo = 1;
+    string id = 2;
+  }
+
+  // TODO: add a picker option to pick commit by uuid.
```
Comment (brendoncarroll) [inline src/pfs/pfs.proto:164]: I think these are used for public documentation.  I'd remove internal comments like TODOs.

**Review by brendoncarroll — APPROVED** [review]

File: src/pfs/pfs.proto [hunk 2]
```diff
@@ -123,7 +151,40 @@ message Commit {
...
+  // the uuid could be generated using AES so that it doesn't have to be stored.
+
+  // This models .N syntax.
+  message BranchRoot {
+    uint32 offset = 1;
```
Comment (brendoncarroll) [inline src/pfs/pfs.proto:169]: This is necessary because it's the opposite direction from `AncestorOf`?

Comment (FahadBSyed) [reply src/pfs/pfs.proto:169]: Yes and its a feature we have here:
https://docs.pachyderm.io/latest/learn/glossary/ancestry-syntax/

**Review by bbonenfant — APPROVED** [review]
This looks good to me. The INT team is probably going to want to write some code to wrap these objects and make these a little nicer to use; similar to what we did [here](https://github.com/pachyderm/pachyderm/blob/master/python-sdk/pachyderm_sdk/api/pfs/_additions.py). I'll make a ticket to track this.

**Pull request merged**